src/verilog/verilog_interfaces.cpp

```diff
@@ -50,17 +50,18 @@ Function: verilog_typecheckt::interface_ports
 
 \*******************************************************************/
 
-void verilog_typecheckt::interface_ports(irept::subt &ports)
+void verilog_typecheckt::check_module_ports(
+  const verilog_module_sourcet &module_source)
 {
-  const irept &module_source=module_symbol.type.find(ID_module_source);
-  const irept &module_ports=module_source.find(ID_ports);
+  const auto &module_ports = module_source.ports();
 
-  ports.resize(module_ports.get_sub().size());
+  auto &ports = module_symbol.type.add(ID_ports).get_sub();
+  ports.resize(module_ports.size());
```

Comment:
How about getting rid of the `resize` here and change it as follows:
1. Use `reserve` here.
2. Get rid of `nr`.
3. Use `push_back` instead of access via `[nr]`.
4. Use `port_names[name]=ports.size() - 1;`

That would avoid any risk of having uninitialised ports in here.

Reply:
ok, will make separate PR